tests/Unit/ParallelAlgorithms/Events/Test_ObserveVolumeIntegrals.cpp

```diff
@@ -0,0 +1,261 @@
+  using type = double;
+};
+
+struct MockContributeReductionData {
+  struct Results {
+    observers::ObservationId observation_id;
+    std::string subfile_name;
+    std::vector<std::string> reduction_names{};
+    double time;
+    double volume;
+    std::vector<double> volume_integrals{};
+  };
+  static Results results;
+
+  template <typename ParallelComponent, typename... DbTags,
+            typename Metavariables, typename ArrayIndex, typename... Ts>
+  static void apply(db::DataBox<tmpl::list<DbTags...>>& /*box*/,
+                    Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const observers::ObservationId& observation_id,
+                    const std::string& subfile_name,
+                    const std::vector<std::string>& reduction_names,
+                    Parallel::ReductionData<Ts...>&& reduction_data) noexcept {
+    results.observation_id = observation_id;
+    results.subfile_name = subfile_name;
+    results.reduction_names = reduction_names;
+    results.time = std::get<0>(reduction_data.data());
+    results.volume = std::get<1>(reduction_data.data());
+    results.volume_integrals = std::get<2>(reduction_data.data());
+  }
+};
+
+// Need forward declaration to avoid linking error.
+MockContributeReductionData::Results MockContributeReductionData::results{};
+
+template <typename Metavariables>
+struct ElementComponent {
+  using component_being_mocked = void;
+
+  using metavariables = Metavariables;
+  using array_index = int;
+  using chare_type = ActionTesting::MockArrayChare;
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+};
+
+template <typename Metavariables>
+struct MockObserverComponent {
+  using component_being_mocked = observers::Observer<Metavariables>;
+  using replace_these_simple_actions =
+      tmpl::list<observers::Actions::ContributeReductionData>;
+  using with_these_simple_actions = tmpl::list<MockContributeReductionData>;
+
+  using metavariables = Metavariables;
+  using array_index = int;
+  using chare_type = ActionTesting::MockArrayChare;
+  using phase_dependent_action_list =
+      tmpl::list<Parallel::PhaseActions<typename Metavariables::Phase,
+                                        Metavariables::Phase::Initialization,
+                                        tmpl::list<>>>;
+};
+
+template <typename System>
+struct Metavariables {
+  using component_list = tmpl::list<ElementComponent<Metavariables>,
+                                    MockObserverComponent<Metavariables>>;
+  using const_global_cache_tags = tmpl::list<>;  //  unused
+  enum class Phase { Initialization, Testing, Exit };
+
+  struct ObservationType {};
+  using element_observation_type = ObservationType;
```

Comment:
I'm not sure if we really need this, and if we do, if `ObserveVolumeIntegrals` should use the same `ObservationType` as `ObserveErrorNorms`. It wouldn't hurt to define an `ObservationType` struct in `ObserveVolumeIntegrals.hpp` instead of retrieving it from the `Metavariables` I think. @nilsdeppe do you have more insight here? If there's doubt, we can just keep it as-is for now.